### PR TITLE
migrate maria to postgresql

### PR DESCRIPTION
**Pull request by wy65701436**

### CODE REVIEW

**Comment by paulczar** [comment]
Is there a design goal to switch from MySQL based to Postgres? This would be a fairly major breaking change that would affect anyone currently running Harbor when they go to upgrade.  We would need to ensure a smooth upgrade process including data migration scripts ( or at least documentation ) from one to the other.

With the potential disruption to existing users, I think we'd need a really good reason to do this as a straight cutover like this, may make more sense to support both for a period of time to give people a chance to move their data.

**Comment by yixingjia** [comment]
@paulczar Thanks a lot for you comment and suggestions.
We plan to migrate to Postgresql due to several reasons.

1> To limit the maintenance effort for our customer.
Currently Harbor has both Mariadb, postgresql, ( For old version of Harbor there are Mysql as well)

2> We have plan to Support HA deployment of Harbor. Setting up different DB clusters are expensive. 

This Patch is only for database driver change.
In the coming patches, we will surely provide data migration scripts for user to transfer from MariaDB to Postgresql. 
We will also document the detail/solid steps for this part. So please don't worry about the upgrade.

Please let us know if you have further more questions/concerns.  

As always thanks a lot for choose Harbor.

**Comment by paulczar** [comment]
:+1:

File: src/common/dao/base.go [hunk 1]
```diff
@@ -87,14 +87,13 @@ func InitDatabase(database *models.Database) error {
...
 func getDatabase(database *models.Database) (db Database, err error) {
 	switch database.Type {
-	case "", "mysql":
-		db = NewMySQL(database.MySQL.Host,
-			strconv.Itoa(database.MySQL.Port),
-			database.MySQL.Username,
-			database.MySQL.Password,
-			database.MySQL.Database)
-	case "sqlite":
-		db = NewSQLite(database.SQLite.File)
```
Comment (yixingjia) [inline src/common/dao/base.go]: Is this "sqlite" section removed by mistake?

Comment (yixingjia) [reply src/common/dao/base.go]: Ok, seem we will not support sqlite

File: src/common/dao/pgsql.go [hunk 2]
```diff
@@ -24,7 +24,7 @@ import (
 
 type pgsql struct {
 	host     string
-	port     int
+	port     string
```
Comment (yixingjia) [inline src/common/dao/pgsql.go]: any reason we change it from int to string?

File: src/common/dao/testutils.go [hunk 3]
```diff
@@ -60,23 +68,7 @@ func PrepareTestForMySQL() {
...
-		log.Fatalf("environment variable SQLITE_FILE is not set")
-	}
-
-	database := &models.Database{
-		Type: "sqlite",
-		SQLite: &models.SQLite{
-			File: file,
-		},
-	}
+	log.Infof("POSTGRES_HOST: %s, POSTGRES_USR: %s, POSTGRES_PORT: %d, POSTGRES_PWD: %s\n", dbHost, dbUser, dbPort, dbPassword)
```
Comment (yixingjia) [inline src/common/dao/testutils.go]: Let's try to avoid log password in any place.

File: src/common/dao/base.go (hunk 1 above)
Comment (wy65701436) [reply src/common/dao/base.go]: yes, will not support sqlite and maria.

**Pull request closed without merging**